### PR TITLE
Fix mixed precision in TF models

### PR DESCRIPTION
**Pull request by jplu**
# What does this PR do?

This PR aims to fix the mixed precision issues when `tf.keras.mixed_precision.experimental.set_policy()` is set to something else than `tf.float32`.  In the same page, this PR aims to fix some TFLite quantization issues.

Before to further continue this PR, the PR #9418 has to be merged.

Fixes # (issue)

#7052

### CODE REVIEW

File: src/transformers/activations_tf.py [hunk 1]
```diff
@@ -56,14 +57,20 @@ def mish(x):
 
 def gelu_fast(x):
     x = tf.convert_to_tensor(x)
-    coeff1 = tf.cast(7978845608, x.dtype)
+    coeff1 = tf.cast(0.7978845608, x.dtype)
```
Comment (patrickvonplaten) [inline src/transformers/activations_tf.py:60]: wow was that wrong the whole time before?

**Review by patrickvonplaten — APPROVED** [review]

File: src/transformers/activations_tf.py (hunk 1 above)
Comment (jplu) [reply src/transformers/activations_tf.py:60]: Yep! I'm surprised that this had not been spotted before 😄

File: src/transformers/activations_tf.py [hunk 2]
```diff
@@ -56,14 +57,20 @@ def mish(x):
...
     coeff2 = tf.cast(0.044715, x.dtype)
 
     return 0.5 * x * (1.0 + tf.tanh(x * coeff2 * (1.0 + coeff1 * x * x)))
 
 
+if version.parse(tf.version.VERSION) >= version.parse("2.4"):
+    gelu = tf.keras.activations.gelu
+else:
+    gelu = tf.keras.layers.Activation(_gelu)
```
Comment (sgugger) [inline src/transformers/activations_tf.py:69]: Nice to be able to use the TF version now!

**Review by sgugger — APPROVED** [review]
LGTM, thanks for fixing this!

**Review by LysandreJik — APPROVED** [review]

**Pull request merged**